backend/src/main/java/com/htecgroup/skynest/service/impl/EmailServiceImpl.java

```diff
@@ -19,17 +21,22 @@
 @Log4j2
 @AllArgsConstructor
 public class EmailServiceImpl implements EmailService {
-  private JavaMailSender javaMailSender;
+
+  private final JavaMailSender javaMailSender;
+  private final TemplateEngine templateEngine;
 
   @Override
   @Async
   public void send(Email email) {
+    Context context = new Context();
+    context.setVariable("args", email.getArgs());
+
     try {
       MimeMessage mimeMessage = javaMailSender.createMimeMessage();
       MimeMessageHelper helper = new MimeMessageHelper(mimeMessage, "utf-8");
-      helper.setText(email.getEmailBody(), email.getHtml());
+      helper.setText(templateEngine.process(email.getEmailType().getTemplate(), context), true);
```

Comment:
everytime when you have same chained method calls, it is good to extract that in separate variable, in this case: email.getEmailType()